joblib/hashing.py

```diff
@@ -129,8 +129,13 @@ def save_global(self, obj, name=None, pack=struct.pack):
     dispatch[type(pickle.dump)] = save_global
 
     def _batch_setitems(self, items):
-        # forces order of keys in dict to ensure consistent hash
-        Pickler._batch_setitems(self, iter(sorted(items)))
+        # forces order of keys in dict to ensure consistent hash.
+        # keys are ordered using their hash.
+        if PY3:
+            Pickler._batch_setitems(self, sorted(items,
+                                                 key=lambda k: hash(k[0])))
+        else:
+            Pickler._batch_setitems(self, iter(sorted(items)))
```

Comment:
Why is it not possible to use the same code under both Python 2 and Python 3? Maybe it's good not to change the Python hash values under Python 2.